ravenframework/Optimizers/GradientDescent.py

```diff
@@ -67,6 +67,12 @@ class GradientDescent(RavenSampled):
                         # TODO change in input space?
                         'objective': r"""provides the maximum relative change in the objective function for convergence.""",
                         'stepSize': r"""provides the maximum size in relative step size for convergence.""",
+                        'iterationLimit': r"""provides the maximum number of iterations at which the optimizer will terminate regardless of the status of any convergence criteria.
+                                              Note that for Gradient Descent this counts only the optimization sweeps not the gradient evaluations.
+                                              It is important to differentiate this from sampler'sinit <limit> node as the latter counts every model evaluation,
+                                              hence, if n variables we will have 2n gradient evaluation if using centeral difference, number of iteration will be <limit>/(2n). If is less than the <iterationlimit> the GA will honor this lower limit.
+                                              I.e., the termination (given all other covergence criteria are not met) will occur at min{<iterationLimit>, ceil(<limit>/gradient evaluations per variable/\# of variables) +1}
+                                              For instance: if the \# of Variables is 2, and we perform 2 evaluations per variable for gradients, and if <limit> = 20, and <iterationLimit> is 10, the termination will occur at min{ceiling(20/(2*2))+1,10} = 6."""
```

Comment:
Some light editing, I suggest:
```tex
provides the maximum number of iterations at which the optimizer will terminate regardless of the status of any 
convergence criteria. Note that for Gradient Descent each new optimal point with its nearby gradient evaluation 
points all count as a single iteration. It is important to differentiate \xmlNode{iterationLimit} from the sampler's 
\xmlNode{limit} node as the latter counts every model evaluation, not just successive steps in the optimization 
algorithm. Hence, if $n$ variables are being optimized, the optimizer perform $n+1$ samples (in 
\texttt{FiniteDifference}) per iteration. If for example we have 4 variables, and the \xmlNode{limit} is set to 100, 
then at most 20 optimization iterations can be performed, since there are 5 model evaluations per optimization 
iteration. If we have 4 variables and instead the \xmlNode{iterationLimit} is set to 100, then at most 100 
optimization iterations can be performed, which results in 500 model evaluations. The optimizer will terminate on 
the first limit it observes. \default{None}.
```
Is this default correct, or is there a default iteration limit?